docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,353 @@
+
+Example: 
+
+`compilerversion\01.2.3-example-sha\0sourceencoding\0utf-8\0checked\01\0unsafe\01\0langversion\0latest\0nullable\0Enable` 
+
+## List of Compiler Flags
+
+#### CSharp Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* deterministic
+* embed
+* filealign
+* highentropyva
+* link
+    - Represented by a metadata reference with `EmbededInteropTypes=true` 
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main
+    - Already stored in PDB as the entry point token
+* moduleassemblyname
+* modulename
+* nostdlib
+    - Will be represented in metadata reference embedded in pdb
+* nowin32manifest
+* pdb
+* platform
+* publicsign
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32res
+
+#### CSharp Flags Not Included
+
+* bugreport
+* delaysign
+* doc
+* errorendlocation
+* errorlog
+* errorreport
+* fullpaths
+* incremental
+* keycontainer
+* keyfile
+* noconfig
+* nologo
+* nowarn
+* out
+* parallel
+* pathmap
+* preferreduilang
+* recurse
+* refout
+* refonly
+* reportanalyzer
+* ruleset
+* utf8output
+* version
+* warn
+* warnaserror
+
+#### Visual Basic Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* filealign
+* highentropyva
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main 
+* nostdlib
+    - Will be represented in metadata reference embedded in pdb
+* platform
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32resource
+
+#### Visual Basic Flags Not Included
+
+* bugreport
+* delaysign
+* doc
+* errorreport
+* help
+* keycontainer
+* keyfile
+* libpath
+* moduleassemblyname
+* modulename
+* netcf
+* noconfig
+* nologo
+* nowarn
+* nowin32manifest
+* optimize
+* optioncompare
+* optionexplicit
+* optioninfer
+* out
+* parallel
+* preferreduilang
+* quiet
+* recurse
+* refonly
+* refout
+* rootnamespace
+* ruleset
+* sdkpath
+* utf8output
+* vbruntime
+* verbose
+* warnaserror
+
+#### Options For CSharp
+
+See [compiler options](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/listed-alphabetically) documentation
+
+| PDB Key | Documentation   |
+| ------- | --------------- |
+| portability-policy | see [portability policy](#portability-policy) |
+| checked | [checked](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/checked-compiler-option) |
+| default-encoding | see [file encoding](#file-encoding) |
+| compilerversion | full version with SHA |
```

Comment:
>compilerversion [](start = 2, length = 15)

Missing `-`.

Reply:
I'm just going to standardize on using `-` everywhere. I think I like it more, since it's more readable to humans.